arviz/stats/diagnostics.py

```diff
@@ -341,40 +341,70 @@ def rhat(data, *, var_names=None, method="rank", dask_kwargs=None):
     )
 
 
-def mcse(data, *, var_names=None, method="mean", prob=None, dask_kwargs=None):
-    """Calculate Markov Chain Standard Error statistic.
+def mcse(
+    data,
+    *,
+    var_names=None,
+    method="mean",
+    prob=None,
+    func=None,
+    mcse_kwargs=None,
+    func_kwargs=None,
+    dask_kwargs=None,
+):
+    r"""Calculate Markov Chain Standard Error statistic.
 
     Parameters
     ----------
-    data : obj
+    data : InferenceData-like or 2D array-like
         Any object that can be converted to an :class:`arviz.InferenceData` object
         Refer to documentation of :func:`arviz.convert_to_dataset` for details
         For ndarray: shape = (chain, draw).
         For n-dimensional ndarray transform first to dataset with ``az.convert_to_dataset``.
-    var_names : list
+    var_names : list of str, optional
         Names of variables to include in the rhat report
-    method : str
-        Select mcse method. Valid methods are:
+    method : {'mean', 'sd', 'median', 'quantile', 'func'}, optional
+        The method to use when estimating the MCSE.
         - "mean"
         - "sd"
         - "median"
         - "quantile"
+        - "func"
 
-    prob : float
+        Methods "mean", "sd", "median" and "quantile" are described in [1]_.
+
+    prob : float, optional
         Quantile information.
+    func : callable, optional
```

Comment:
we could also consider allowing some strings here. e.g. using `"circmean"` expands to `stats.circmean` as `func` here and also fills the `mcse_kwargs` with `{"var_func": stats.circvar}`